cherrypy/lib/auth_digest.py

```diff
@@ -445,6 +502,8 @@ def digest_auth(realm, get_ha1, key, debug=False, accept_charset='utf-8'):
         A secret string known only to the server, used in the synthesis
         of nonces.
     """
+    server_algorithms = valid_algorithms.intersection(algorithms)
```

Comment:
Let's just use this equivalent syntax:
```suggestion
    server_algorithms = valid_algorithms & algorithms
```